inject/src/main/java/io/micronaut/inject/annotation/AnnotationMetadataHierarchy.java

```diff
@@ -322,15 +322,18 @@ public String[] stringValues(@NonNull Class<? extends Annotation> annotation, @N
     @NonNull
     @Override
     public String[] stringValues(@NonNull String annotation, @NonNull String member) {
-        String[] values = hierarchy[0].stringValues(annotation, member);
-        for (int i = 1; i < hierarchy.length; i++) {
-            AnnotationMetadata annotationMetadata = hierarchy[i];
-            final String[] moreValues = annotationMetadata.stringValues(annotation, member);
-            if (ArrayUtils.isNotEmpty(moreValues)) {
-                values = ArrayUtils.concat(values, moreValues);
+        return firstNotEmptyStringValuesInHierarchy(annotation, member);
+    }
+
+    @NonNull
+    private String[] firstNotEmptyStringValuesInHierarchy(@NonNull String annotation, @NonNull String member) {
```

Comment:
I think the previous code is more correct, all the values are aggregated.

Reply:
Please note that with: 

```
     @Controller("/example")
    @RolesAllowed("isAuthenticated()")
    static class ExampleController {

        @Get("/admin")
        @RolesAllowed({"ROLE_ADMIN", "ROLE_X"})
        void  admin(HttpRequest<?> request) {
        }
```

`stringValues` returns  `"isAuthenticated()", "ROLE_ADMIN", "ROLE_X"` while getValues returns `"ROLE_ADMIN", "ROLE_X"`. The latter looks correct to me.